src/workloads/execution/SecondaryReadsGenny.yml

```diff
@@ -0,0 +1,252 @@
+        Active: {^Parameter: {Name: "ActivePhases", Default: [-1]}}
+        NopInPhasesUpTo: *max_phase
+        PhaseConfig:
+          Repeat: 1
+            
+
+- TemplateName: WritesOnPrimaryTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: WritesOnPrimary}}
+    Type: CrudActor
+    Threads: *writer_threads
+    Database: *db
+    Phases:
+      OnlyActiveInPhases:
+        Active: {^Parameter: {Name: "ActivePhases", Default: [-1]}}
+        NopInPhasesUpTo: *max_phase
+        PhaseConfig:
+          Duration: *test_duration
+          Collection: Collection0
+          Operations:
+          - OperationName: insertOne
+            OperationCommand:
+              Document:
+                x: {^RandomInt: {min: *initial_nb_docs, max: 1000000000}}
+                #s: {^Repeat: {count: *document_size, fromGenerator: {^Choose: {from: ["x"]}}}} --> why does this not work?
+                s: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+
+- TemplateName: ReadsOnSecondaryTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: ReadsOnSecondary}}
+    Type: CrudActor
+    Threads: {^Parameter: {Name: "Threads", Default: -1}}
+    ClientName: Secondary
+    Database: *db
+    Phases:
+      OnlyActiveInPhases:
+        Active: {^Parameter: {Name: "ActivePhases", Default: [-1]}}
+        NopInPhasesUpTo: *max_phase
+        PhaseConfig:
+          Duration: *test_duration
+          Collection: Collection0
+          Operations:
+          - OperationName: find
+            OperationCommand:
+              Filter: { x: {$gte: {^RandomInt: {min: 0, max: *initial_nb_docs}}}} # this should be *initial_nb_docs - *read_limit but we don't seem to have and add/subtract generator and it doesn't really impact performance
+              Options:
+                Limit: *read_limit
+                BatchSize: 200
+                ReadPreference:
+                  ReadMode: secondaryPreferred
+
+- TemplateName: ReplicationLagTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: ReplicationLag}}
+    Type: CrudActor
+    Threads: 1
+    Database: *db
+    Phases:
+      OnlyActiveInPhases:
+        Active: {^Parameter: {Name: "ActivePhases", Default: [-1]}}
+        NopInPhasesUpTo: *max_phase
+        PhaseConfig:
+          Repeat: 1
+          Collection: Collection0
+          Operations:
+          - OperationName: insertOne
+            OperationCommand:
+              Document: {s: 1}
+              OperationOptions:
+                WriteConcern:
+                  Level: 3
```

Comment:
This deserves a comment at top that this has to run with 3 nodes. I wonder if we want w:majority instead, as that will be more forgiving, and it will give the majority lag time.

Reply:
This needs to be Writeconcern == NumberOfNodes since we only have the additional reads on one secondary so with w:majority this would not correctly measure the replication lag of this node under the additional stress of the reads.

I'll add a comment at the top that this needs to be run in a 3-node replica set. The smooth solution would be to dynamically refer to the size of the cluster here, then it could also run in a 2-node replica set, do you know of any way to do this?

Reply:
Note: waiting on adding the comment to see if there is a better solution

Reply:
I do not know a good way to do that in Genny. In the queisce logic I think we explicitly query for how many nodes there are in the replica set, and use that value. 

There is a real `w:majority` write concern, but unfortunately no `w:all` write concern (I wish there was). 
I suppose we could make the write concern an parameter to the workload, and then we could override it to set it. I know that is possible, but it would take me some work to do it myself. I think we have a recent example though passing in a parameter.

Reply:
@jimoleary do you have thoughts on the above to do a `w:all` write and not explicitly know the replica set size in this file?

Reply:
I don't know of any magic or cleverness to get around this. **At the end of the day, we are still hamstrung by the inability to pass parameters into a genny workload file from dsi.** 

The only way of stopping them being run on the wrong mongo setup is to ensure the value is set correctly for the mongodb_setup autorun. 

Making the value a parameter with the correct default is probably a good start anyway.

```
              OperationOptions: {^Parameter: {Name: "OperationOptions", Default: {WriteConcern:{Level: 3, Journal: true}}}}
```

I think that [LargeRange.yml](https://github.com/10gen/PrivateWorkloads/blob/production/src/workloads/customer_representative/coinbase/balance/LargeRange.yml) calling [Large.ym](https://github.com/10gen/PrivateWorkloads/blob/production/src/workloads/customer_representative/coinbase/balance/Large.yml) is the closed to what you may be alluding to. 

But storing these actors in the phases directory (like [this](https://github.com/mongodb/genny/blob/9354506965654f4fd23ba61d1c7ccb1163b145d3/src/phases/scale/DesignDocWorkloadPhases.yml) and loading them like [this](https://github.com/mongodb/genny/blob/0c6262b06e8f543d69de8ac271359dbb7c82fd20/src/workloads/scale/UpdateMillionDocsInTxn.yml#L14-L29) might be better.